tmt/utils.py

```diff
@@ -3657,32 +3657,86 @@ def _create_directory() -> tuple[str, Optional[Exception]]:
 
 
 def create_file(
+        *,
         path: Path,
         content: str,
         name: str,
         dry: bool = False,
         force: bool = False,
         mode: int = 0o664,
-        quiet: bool = False) -> None:
-    """ Create a new file, handle errors """
-    say = log.debug if quiet else echo
-    action = 'would be created' if dry else 'created'
-    if path.exists():
-        if force:
-            action = 'would be overwritten' if dry else 'overwritten'
-        else:
-            raise FileError(f"File '{path}' already exists.")
+        quiet: bool = False,
+        logger: tmt.log.Logger) -> None:
+    """
+    Create a new file.
+
+    Before creating the file, function checks whether it exists
+    already - the existing file is **not** removed and re-created,
+    unless ``force`` is set.
+
+    The outcome of the operation will be logged in a debug log, but
+    may also be sent to console with ``quiet=False``.
+
+    :param path: a path to be created.
+    :param content: content to save into the file
+    :param name: a "label" of the path, used for logging.
+    :param dry: if set, the file would not be created or overwritten. Still,
+        the existence check will happen.
+    :param force: if set, the file would be overwritten if it already exists.
+    :param mode: permissions to set for the file.
+    :param quiet: if set, an outcome of the operation would be logged
+        to console.
+    :param logger: logger to use for logging.
+    :raises FileError: when function tried to create the file,
+        but failed.
+    """
+
+    # Streamline the logging a bit: wrap the creating with a function returning
+    # a message & optional exception. Later we will send the message to debug
+    # log, and maybe also to console.
+    def _create_file() -> tuple[str, Optional[Exception]]:
+        # When overwriting an existing path, we need to provide different message.
+        # Let's mark the existence down for later.
+        overwriting = False
 
-    if dry:
-        say(f"{name.capitalize()} '{path}' {action}.")
+        if path.exists():
+            if not force:
+                message = f"{name.capitalize()} '{path}' already exists."
+
+                # Return a custom exception - it was not raised by any FS-related code,
+                # but we need to signal the operation failed to our caller.
+                return message, FileExistsError(message)
+
+            if dry:
+                return f"{name.capitalize()} '{path}' would be overwritten.", None
+
+            overwriting = True
```

Comment:
how about instead of having bool overwriting, you can use string action with default on 'created' and replacing it here to 'overwritten' and use it in outputs in `if dry` for these 2, thus merging them to just be the following one

and reusing later on lines 3724-3727 which would again make it just one return with no need to check anything

Reply:
Could you write how it would look like, in a pseudocode or Python?

Reply:
```action = 'created'
if path.exists():
  action = 'overwritten'
if dry:
  return f"{name.capitalize()} '{path}' would be {action}.", None
try:
  # overwrite or write
return f"{name.capitalize()} '{path}' {action}.", None
```

Reply:
Refactored in 7d878979

Reply:
@KwisatzHaderach pls re-review